stats/otel.go

```diff
@@ -61,37 +68,60 @@ func (s *otelStats) Start(ctx context.Context, goFactory GoRoutineFactory) error
 		return fmt.Errorf("failed to create open telemetry resource: %w", err)
 	}
 
-	options := []otel.Option{otel.WithInsecure()} // @TODO: could make this configurable
+	options := []otel.Option{otel.WithInsecure(), otel.WithLogger(s.logger)}
 	if s.otelConfig.tracesEndpoint != "" {
 		options = append(options, otel.WithTracerProvider(
 			s.otelConfig.tracesEndpoint,
 			s.otelConfig.tracingSamplingRate,
 		))
 	}
-	if s.otelConfig.metricsEndpoint != "" {
-		meterProviderOptions := []otel.MeterProviderOption{
-			otel.WithMeterProviderExportsInterval(s.otelConfig.metricsExportInterval),
-		}
-		if len(s.config.defaultHistogramBuckets) > 0 {
+
+	meterProviderOptions := []otel.MeterProviderOption{
+		otel.WithMeterProviderExportsInterval(s.otelConfig.metricsExportInterval),
+	}
+	if len(s.config.defaultHistogramBuckets) > 0 {
+		meterProviderOptions = append(meterProviderOptions,
+			otel.WithDefaultHistogramBucketBoundaries(s.config.defaultHistogramBuckets),
+		)
+	}
+	if len(s.config.histogramBuckets) > 0 {
+		for histogramName, buckets := range s.config.histogramBuckets {
 			meterProviderOptions = append(meterProviderOptions,
-				otel.WithDefaultHistogramBucketBoundaries(s.config.defaultHistogramBuckets),
+				otel.WithHistogramBucketBoundaries(histogramName, defaultMeterName, buckets),
 			)
 		}
-		if len(s.config.histogramBuckets) > 0 {
-			for histogramName, buckets := range s.config.histogramBuckets {
-				meterProviderOptions = append(meterProviderOptions,
-					otel.WithHistogramBucketBoundaries(histogramName, defaultMeterName, buckets),
-				)
-			}
+	}
+	if s.otelConfig.metricsEndpoint != "" {
+		options = append(options, otel.WithMeterProvider(append(meterProviderOptions,
+			otel.WithGRPCMeterProvider(s.otelConfig.metricsEndpoint),
+		)...))
+	} else if s.otelConfig.enablePrometheusExporter {
+		if s.otelConfig.prometheusMetricsPort < 1 || s.otelConfig.prometheusMetricsPort > 65535 {
+			return fmt.Errorf("invalid prometheus metrics port %d", s.otelConfig.prometheusMetricsPort)
 		}
-		options = append(options, otel.WithMeterProvider(s.otelConfig.metricsEndpoint, meterProviderOptions...))
+		options = append(options, otel.WithMeterProvider(append(meterProviderOptions,
+			otel.WithPrometheusExporter(prometheus.DefaultRegisterer),
```

Comment:
It would be nice to be able to override the `DefaultRegisterer`

Reply:
Sure, that's easy to add :+1:

Reply:
@atzoum done [here](https://github.com/rudderlabs/rudder-go-kit/pull/13/commits/a129de81672156464b7c472047ba383ab12ad520).